x/nft/blockchain/init.go

```diff
@@ -0,0 +1,69 @@
+package blockchain
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/pkg/errors"
+)
+
+// Initializer fulfils the InitStater interface to load data from
+// the genesis file
+type Initializer struct {
+}
+
+var _ weave.Initializer = (*Initializer)(nil)
+
+// FromGenesis will parse initial tokens information from the genesis and
+// persist it in the database.
+func (i *Initializer) FromGenesis(opts weave.Options, db weave.KVStore) error {
+	var nfts struct {
+		Blockchains []genesisBlockchainToken `json:"blockchains"`
+	}
+	if err := opts.ReadOptions("nfts", &nfts); err != nil {
+		return errors.Wrap(err, "read options")
+	}
+	return setTokens(db, nfts.Blockchains)
+}
+
+type genesisBlockchainToken struct {
+	ID    string `json:"id"`
+	Owner string `json:"owner"`
+	Chain struct {
+		ChainID      string `json:"chain_id"`
+		NetworkID    string `json:"network_id"`
+		Name         string `json:"name"`
+		Enabled      bool   `json:"enabled"`
+		Production   bool   `json:"production"`
+		MainTickerID string `json:"main_ticker_id"`
+	} `json:"chain"`
+	IOV struct {
+		Codec       string `json:"codec"`
+		CodecConfig string `json:"codec_config"`
+	} `json:"iov"`
+}
+
+func setTokens(db weave.KVStore, tokens []genesisBlockchainToken) error {
+	bucket := NewBucket()
+	for _, t := range tokens {
+		chain := Chain{
+			ChainID:      t.Chain.ChainID,
+			NetworkID:    t.Chain.NetworkID,
+			Name:         t.Chain.Name,
+			Enabled:      t.Chain.Enabled,
+			Production:   t.Chain.Production,
+			MainTickerID: []byte(t.Chain.MainTickerID),
+		}
+
+		iov := IOV{
+			Codec:       t.IOV.Codec,
+			CodecConfig: t.IOV.CodecConfig,
+		}
+		obj, err := bucket.Create(db, weave.Address(t.Owner), weave.Address(t.ID), nil, chain, iov)
```

Comment:
What kind of data goes into `owner` and `ID`? Does weave.Address hexdecode? Addresses in th genesis file should be hex consistently (like in `wallets`)

Reply:
I cannot find any validation of neither `owner` nor `id`. Given value is not being converted or deserialized, but used as is. It's cast directly to `weave.Address`, which is raw bytes representation of the address.

If it makes it easier for you, I could make it accept only hex encoded value. I think that would make it easier to use and allow to ensure the data passed is right. What do you think?

Reply:
Yes, JSON should hold hex encoded addresses. Otherwise we cannot use this because I cannot insert 20 raw bytes into JSON strings

Reply:
:+1:
I will create a pull request for that change.